src/test/java/testresults/CVSB_8778/SubmitTestAdrNewScreenHgvAndTrl_CVSB_8778.java

```diff
@@ -0,0 +1,188 @@
+package testresults.CVSB_8778;
+
+import net.serenitybdd.junit.runners.SerenityRunner;
+import net.thucydides.core.annotations.Steps;
+import net.thucydides.core.annotations.Title;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import steps.*;
+import steps.composed.TestTypeCategoryComp;
+import util.BaseTestClass;
+
+@RunWith(SerenityRunner.class)
+public class SubmitTestAdrNewScreenHgvAndTrl_CVSB_8778 extends BaseTestClass {
+
+    @Steps
+    TestSteps testSteps;
+
+    @Steps
+    TestTypeCategoryComp testTypeCategoryComp;
+
+    @Steps
+    TestTypeDetailsSteps testTypeDetailsSteps;
+
+    @Steps
+    TestTypeSubcategorySteps testTypeSubcategorySteps;
+
+    @Steps
+    TestTypeCategorySteps testTypeCategorySteps;
+
+    @Steps
+    CountryOfRegistrationSteps countryOfRegistrationSteps;
+
+    @Steps
+    EUVehicleCategorySteps euVehicleCategorySteps;
+
+    @Steps
+    OdometerReadingSteps odometerReadingSteps;
+
+    @Steps
+    TestReviewSteps testReviewSteps;
+
+    @Steps
+    PreparerSteps preparerSteps;
+
+    @Steps
+    AdrDetailsStep adrDetailsStep;
+
+    @Steps
+    SiteVisitSteps siteVisitSteps;
+
+    @Steps
+    IdentifyVehicleSteps identifyVehicleSteps;
+
+    @Steps
+    VehicleDetailsSteps vehicleDetailsSteps;
+
+    @Steps
+    TestHistorySteps testHistorySteps;
+
+    @Steps
+    TestHistoryDetailsSteps testHistoryDetailsSteps;
+
+    @Steps
+    ConfirmationPageSteps confirmationPageSteps;
+
+    @Steps
+    TrailerDetailsSteps trailerDetailsSteps;
+
+    @Steps
+    IdentifyTrailerSteps identifyTrailerSteps;
+
+    @Title("CVSB-8778 - AC1 - Rule for dynamic 'Battery list' fields - TC3 - AC3 - Field 'Applicable' = Yes")
+    @Test
+    public void testAdrDetailsFieldsSemiTrailerBatteryApplicable() {
+        testTypeCategoryComp.goToTestPageBySelectingASpecificTrailer("777776");
+        preparerSteps.startTest();
+        preparerSteps.confirmInPopUp();
+        testSteps.checkTrailerTestDetails("C777776", "ABCDEFGH777776");
+        testSteps.selectTrailerDetails();
+        trailerDetailsSteps.scrollTheDetailPage();
+        trailerDetailsSteps.selectAdrDetails();
+        adrDetailsStep.waitUntilPageIsLoaded();
+        adrDetailsStep.checkAdrDetailSectionIsDisplayed("ADR DETAILS");
+        adrDetailsStep.checkAdrDetailHasValue("Vehicle type", "semi trailer battery");
+        adrDetailsStep.scrollTheAdrDetailPage();
+        adrDetailsStep.checkAdrDetailSectionIsDisplayed("BATTERY LIST");
+        adrDetailsStep.checkAdrDetailSectionContains("BATTERY LIST", "Applicable");
+        adrDetailsStep.checkAdrDetailHasValue("Applicable", "Yes");
+        adrDetailsStep.checkAdrDetailSectionContains("BATTERY LIST", "Reference number");
+        adrDetailsStep.checkAdrDetailHasValue("Reference number", "H903");
+    }
```

Comment:
**Acceptance criteria:**
WHEN there is no data for the "Reference number" field
THEN the field would still be displayed with the copy "Not entered"
is not covered under this test case

Reply:
fixed, updated the seeding data